kyaml/runfn/runfn.go

```diff
@@ -46,46 +56,141 @@ func (r RunFns) Execute() error {
 	// default the containerFilterProvider if it hasn't been override.  Split out for testing.
 	(&r).init()
 
-	// identify the configuration functions in the directory
+	fltrs, err := r.getFilters()
+	if err != nil {
+		return err
+	}
+
+	return r.runFunctions(fltrs)
+}
+
+func (r RunFns) getFilters() ([]kio.Filter, error) {
+	var fltrs []kio.Filter
+
+	// implicit filters from the input Resources
+	f, err := r.getFunctionsFromInput()
+	if err != nil {
+		return nil, err
+	}
+	fltrs = append(fltrs, f...)
+
+	// explicit filters from a list of directories
+	f, err = r.getFunctionsFromDirList()
+	if err != nil {
+		return nil, err
+	}
+	fltrs = append(fltrs, f...)
```

Comment:
Functions found implicitly as well as defined in FunctionPaths will run twice. I'm fine with that if you are though.

Reply:
Ah, this makes sense.  I've updated it to default `NoFunctionFromInput` to true if it is unspecified and `len(FunctionPaths) > 0`.  This gives the caller explicit control over how functions are discovered, but provides an unsurprising default behavior -- if you explicitly specify functions we don't discover additional ones by default.